doc/versioning.rst

```diff
@@ -37,11 +37,11 @@ Increment the:
    incompatible API changes, including removing APIs.
 
 2. MINOR version when adding minor functionality and API changes
-   that are backwards compatible.
+   that are backwards compatible. Supported values: 0-99 [1]
 
 3. REVISION when making backwards compatible bug fixes or any editing
    change in the document, including minor changes such as correcting
-   typos. Initial REVISION is 1.
+   typos. Initial REVISION is 1. Supported values: 1-99 [1]
```

Comment:
Is it even necessary to track revisions (in other words, editorial changes) in any implementation version macros? My understanding is that any changes that affect API semantics (including observable side effects) require at least the minor version update.

Reply:
The existing text from the spec says:
> Increment the:
> - MAJOR version when adding major new functionality and making incompatible API changes, including removing APIs.
> - MINOR version when adding minor functionality and API changes that are backwards compatible.
> - REVISION when **making backwards compatible bug fixes** or any editing change in the document, including minor changes such as correcting typos. Initial REVISION is 1.

My understanding is that the existing spec allows incrementing revision number for bug fixes, while major and minor updates may contain new functionality. This RFC does not modify this behavior.

If we want to remove the revision from macros, then the revision field should explicitly define only documentation changes, without the **bold text** above. Since this will require a separate RFC, I consider this out-of-scope for this proposal.

Reply:
This document describes versioning of the specification, therefore in the context of it "bug fixes" mean changes in the specification, and not in any of the implementations. Therefore, I do not think it says about any new functionality - since bugs fixed in the specification obviously do not involve any source code changes.

As described above, if the specification change includes a new minor functionality or compatible API changes, then the minor version needs to be updated. Therefore I remain positive that new revisions only contain editorial changes but no changes to the API semantics.

I am happy to be corrected, ideally with practical examples. Tagging @rscohn2.

Reply:
I am okay to remove the revision part from the proposal if that was the intended use case for the revision field and is currently being followed. I still think the bold text can be improved to have less ambiguity.

Reply:
In 4 years, I have only used revision for formatting/editing. I can remove the "compatible buf fixes" clause.

Reply:
Thank you @rscohn2 . In that case, I will update the RFC to include only major and minor versions in the macro.